### DISCUSSION TITLE
Build exception with different env

### DISCUSSION THREAD

**Discussion by rdrigos** [prisma/prisma#23992] [Q&A] [answered]
### Question

Good afternoon, I'm starting to use PRISMA ORM in my TypeScript code, and I have some questions...
After initializing a project with PRISMA using the command npx prisma init, it creates the following structure:

```
project
    |---- prisma
    |            |---- prisma.schema
    |
    |---- .env
    |---- src (my folder with API files, bots....)
    |---- package.json
    |---- package-lock.json
```
and I would like to move this .env file to the env folder, so it would look like this:

```
project
  |---- env
  |          |---- .env
  |
  |---- prisma
  |          |---- prisma.schema
  |
  |---- .env
  |---- src (my folder with API files, bots....)
  |---- package.json
  |---- package-lock.json
```
without causing an error in prisma.schema at the time of table creation and execution of the PRISMA code. I also would like to show a better structure, how the code will be executed on a local server:

```
project
  |---- env
  |          |---- env.development
  |          |---- env.production
  |
  |---- prisma
  |          |---- prisma.schema
  |
  |---- .env
  |---- src (my folder with API files, bots....)
  |---- package.json
  |---- package-lock.json
```
and in this way, when building:

```
project
  |---- dist
  |         |---- development
  |         |---- production
  |
  |---- env
  |          |---- env.development
  |          |---- env.production
  |
  |---- prisma
  |          |---- prisma.schema
  |
  |---- .env
  |---- src (my folder with API files, bots....)
  |---- package.json
  |---- package-lock.json
```
each of these environments knows which env file to use from the env folder and executes correctly.

**Comment by janpio** [comment]
Prisma only looks for `.env` in the root folder. But you can use `dotenv-cli` or similar packages to load the `.env` file from any location you want.

**Reply by rdrigos** [reply]
Ok, could you explain this to me better? Or leave an article explaining how it works

**Reply by ludralph** [reply]
Hi @rdrigos, Can you take a look at this [section](https://www.prisma.io/docs/orm/more/development-environment/environment-variables/managing-env-files-and-setting-variables#manage-env-files-manually) of the documentation?

**Reply by rdrigos** [reply]
Good morning.

I have this question regarding Prisma ORM. I want to create tables in SQL Server with table schemas, e.g., ATRT for development tables and PROD for production.

I wish that when I run ```npx prisma db migrate```, it creates tables like PROD.users and ATRT.users to avoid issues with production data.... How could I achieve this?

**Reply by ludralph** [reply]
Hi @rdrigos 

To create tables with different schemas in SQL Server using Prisma ORM, you can use the `@@schema` attribute in your Prisma schema. This feature allows you to specify different database schemas for your models. Here's how you can achieve this:
First, enable the multiSchema preview feature in your Prisma schema:
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
  schemas  = ["PROD", "ATRT"]
}
// Then, define your models with the appropriate schema:

model User {
  id    Int    @id @default(autoincrement())
  email String @unique
  name  String?

  @@schema("PROD")
}

model DevelopmentUser {
  id    Int    @id @default(autoincrement())
  email String @unique
  name  String?

  @@schema("ATRT")
}
```
When you run npx prisma migrate dev, Prisma will generate migrations that create tables in the specified schemas. https://www.prisma.io/docs/orm/prisma-schema/data-model/multi-schema

**Comment by ludralph** [comment, marked as answer]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
